src/librustc_lint/builtin.rs

```diff
@@ -687,6 +687,18 @@ impl EarlyLintPass for BadRepr {
     fn check_attribute(&mut self, cx: &EarlyContext, attr: &ast::Attribute) {
         if attr.name() == "repr" {
             let list = attr.meta_item_list();
+            let outer = match attr.style {
+                ast::AttrStyle::Outer => true,
+                ast::AttrStyle::Inner => false,
```

Comment:
Since we all seem to be in agreement this isn't actually allowed, I suggest an `assert_eq!` and simplifying the code accordingly.